libs/datasets/timeseries.py

```diff
@@ -725,22 +730,24 @@ def timeseries_rows(self) -> pd.DataFrame:
         wide_dates = wide_dates.loc[:, wide_dates.columns[-1::-1]]
         wide_dates = wide_dates.rename_axis(None, axis="columns")
 
-        if not self.provenance.empty:
-            # Add one or more columns with the provenance tag(s) for each timeseries
-            # (identified by a <location_id, variable> pair).
-            # From https://stackoverflow.com/a/38369722
-            provenance_columns = (
-                self.provenance.groupby([CommonFields.LOCATION_ID, PdFields.VARIABLE])
-                .apply(lambda df: df.reset_index(drop=True))
-                .unstack()
-                # The first provenance column has the same name as when there was only support
-                # for a single column. Extra columns are named "provenance-1", "provenance-2", ...
-                # These column names are parsed in `read_from_pointer`.
-                .rename(columns=lambda i: TagType.PROVENANCE + ("" if i == 0 else f"-{i}"))
-            )
-            return pd.concat([provenance_columns, wide_dates], axis=1)
-        else:
-            return wide_dates
+        # Each element of output_series will be a column in the returned DataFrame. There is at
+        # least one column for each tag type in self.tag. If a timeseries has multiple tags with
+        # the same type additional columns are added.
```

Comment:
Instead of having one column per annotation, could we just combine them into a single json list? 

thinking about it a bit more - i think this approach would be easier if we represented tags in the dataframe as actual `TagInTimeseries`  objects rather than json encoded strings.  Is there much of an advantage to keeping it as strings in the pandas objects?

Reply:
Fair point. There is a tradeoff between trying to keep data in structures that are well supported by pandas vs customized to exactly what we want. Values in panda DataFrame cells can be operated on with native fast panda operations instead of us needing to write code. I like being able to do most of joining and making subsets of datasets with panda operations. Keeping the disk and memory representations pretty close simplifies the reading and writing code.

Reply:
Another advantage of using separate columns of strings in CSVs on disk is that I currently depend pretty heavily on csvkit for debugging data issues, which will be less effective with heavier use of JSON in CSVs. I'm hoping standing up a dashboard will reduce how much time we spend handling CSVs and give us more flexibility to use other representations.